pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -283,12 +284,43 @@ impl UsedType {
             original => (original, false, None, false),
         };
 
-        Ok(Self { original_ty, resolved_ty, optional, result, variadic, default, composite_type })
+        // if the Type is like `Result<T, E>`, this finds the `T`
+        let mut resolved_ty_inner: Option<syn::Type> = None;
+        if result {
+            if let syn::Type::Path(tp) = &resolved_ty {
+                if let Some(first_segment) =
+                    tp.path.segments.first().map(|segment| Some(segment)).unwrap_or(None)
+                {
+                    if let syn::PathArguments::AngleBracketed(ab) = &first_segment.arguments {
+                        if let Some(first_arg) =
+                            ab.args.first().map(|arg| Some(arg)).unwrap_or(None)
```

Comment:
Hmm, I guess this just assumes that in `Foo<T, U, V, W, ...>` we care about `T` and that's it. It's not ideal but that is true for Result, Option, etc, so I suppose this is fine for now and a strict improvement.

Reply:
Correct. And it only does this for Results. 

Impossible to know type_ids for a Result when you allow the E to be anything.

Reply:
Ah, yeah, sorry, I missed this was only for Result.